Improved footer menu with 3 shortcut buttons and one drop-up for all other actions

strindhaug opened the pull request:
Another example of a slightly more radical approach to fixing https://github.com/wagtail/wagtail/issues/2214
implementing some of the ideas I mentioned in my comments on the less radical https://github.com/wagtail/wagtail/pull/3241 pr.

This solution is a lot more compact and responsive even though it has an additional button.

Current footer, for width comparison (note that the preview button is actually wider than it appears):
![orig](https://cloud.githubusercontent.com/assets/2144849/21424754/179847ae-c845-11e6-9948-57149a621686.png)

New footer in the same width (the preview button now doesn't have extra width since it's not a dropdown anymore; extra preview modes are shown in the common dropdown):
![wide](https://cloud.githubusercontent.com/assets/2144849/21424783/3673c81a-c845-11e6-825b-e04781d82773.png)

Some of the meta-data is hidden (and revisions link is moved to the menu), to allow narrowers screen without wrapping:
![narrow](https://cloud.githubusercontent.com/assets/2144849/21424804/5815e3ea-c845-11e6-8349-697bcccefca8.png)

The 3 direct buttons with icons (save draft, publish, preview) become icon only on smaller widths to avoid wrapping on phone sizes:
![phone](https://cloud.githubusercontent.com/assets/2144849/21424809/5bfd00ba-c845-11e6-8fce-713629630ea3.png)

Color coded (only on hover) delete and unpublish with red and yellow respectively:
![color_code_on_hover](https://cloud.githubusercontent.com/assets/2144849/21424858/b3addb22-c845-11e6-8118-7103ab8088c2.png)

The "Submit for moderation" button  replaces the "Publish" button when the user is not allowed to publish:
![cant_pub](https://cloud.githubusercontent.com/assets/2144849/21424875/d6ae9ce2-c845-11e6-927b-f3409b0e2c6d.png)
It's deliberately not color coded green like the publish button to avoid confusion.

**NOTE** The css is currently hardcoded in a `<style>` block in the template for easy development and there is a lot of commented code for reference while developing. This needs to be cleaned up and the styles must be moved to the sass files before this can be merged.
But even though this code is somewhat ugly; it *is* functional and can be checked out locally for testing.

strindhaug commented:
I changed my mind, the submit for moderation button is now green as well. I also added the mail icon as a temporary icon so that it can hide the very long button text on mobile.

Added lock icon and hiding of text when locked as well, to prevent wrapping.

tmsndrs commented:
I think this is an improvement on the previously suggested design. I worry that it feels a little cluttered though. 

- I think the preview should be separated out from the menu buttons though as it feels less related to editorial workflow. 
- The tick and green of the publish button could communicate that the page is published rather than will be published on click
- I'm not sure about the visual styling of the menu on desktop. It's feels dislocated from the button.
- Had the same thought about publish being replaced by 'submit for moderation' when appropriate. Think this is a good idea. 
- The hover colours don't feel like they're part of the Wagtail theme IMO.

strindhaug commented:
I've experiented some more with the pupup design (haven't committet yet), using a "speech bubble" style arrow to point down to the trigger button instead:

![restyled_popup_menu](https://cloud.githubusercontent.com/assets/2144849/21674010/1c6f31e2-d32c-11e6-9c5a-ece773a651f3.png)

Though it's technically even more separated from the trigger, the arrow-ish design makes it feels more attached somehow.

I agree the preview button feels like it's misplaced, so I moved it to the popup menu saving even more space.

I've added a filled document icon as a temporary placeholder for a save symbol, and a fat chevron icon rotated 90 degrees with css transform as a temporary submit icon.
Would be nice if someone could create some good icons for these two actions and add it to the icon set. If they're sufficiently visually distinct the buttons don't need to be in different color.

The hover colors (not shown) on delete and unpublish is still there. They are actually unmodified colors from the styleguide. The buttons have the "button no" classes and "button warning" classes respectively. I've just overridden the un-hovered color to match normal buttons because they look a bitt garish next to each other. I don't mind removing them if there is strong opposition to using them in a popup; but I personally think they're a nice usability addition making the "destructive" actions more visible when hovered (I know there is a confirmation screen for these actions, but it's annoying to accidentally go to this screen when you were aiming for preview)

strindhaug commented:
Another way of making the buttons easier to differentiate without having to read them is spacing them a bit more when there is available space. 

Using css flex layout this could look like this:
Desktop:
![flex_spaced_desktop](https://cloud.githubusercontent.com/assets/2144849/21675857/a1f3bc78-d333-11e6-977a-f04d310d2161.png)
Mobile:
![flex_spaced_mobile](https://cloud.githubusercontent.com/assets/2144849/21675861/a54c004c-d333-11e6-8d8e-e1fff7148fc9.png)

Since it's wrapping flex box it doesn't require lots of space, while still uses all available space when the screen is wide enough. (And if flex isn't supported in the browser it falls back to normal left floated layout.)

tomdyson commented:
@strindhaug thank you very much for this thoughtful contribution and discussion. There's a lot for us to think about here, in combination with @tmsndrs's proposals at #3269. I would also like to include @teneightfive who has been doing some user testing on these elements as part of the https://github.com/nhsuk/nhsuk-content-store project.

tmsndrs commented:
Any objections to closing this? We're going to take this on from ticket #3269

thibaudcolas commented:
Based on #3269 I can see this is still at the design stage, so will close this WIP PR for now. Can always be re-opened depending on the direction things take 🙂.
